lib/improver/spotdata/read_input.py

```diff
@@ -144,14 +142,22 @@ def get_additional_diagnostics(diagnostic_name, diagnostic_data_path,
                       data, with a single entry is time_extract is provided.
 
     """
-    cubes = Load('multi_file').process(
-        diagnostic_data_path + '/*/*' + diagnostic_name + '*',
-        None)
+    # Search directory structure for all files relevant to current diagnostic.
+    files_to_read = [
```

Comment:
I think this is likely to change radically when spotdata is driven from Improver output, with the structure of storing the output data not (as far as I know) yet decided, but potentially likely to look like the archive? If we are talking hour by hour diagnostic files then it will be more sensible to break up the reading of files by forecast time. At the moment it lazily (in a biggus sense) reads all the cubes and then extracts the relevant sections for each forecast time being considered. I'm struggling, within the broader context of how these routines are configures, to think of a nice way of doing this that doesn't walk a directory tree from the given base directory.